### PR TITLE
feat(destinations): Add `migrate_mode`

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Related to https://github.com/cloudquery/cloudquery/issues/6600, needed for https://github.com/cloudquery/cloudquery/pull/6763

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: specs/destination.go [hunk 1]
```diff
@@ -11,15 +11,18 @@ import (
...
-	WriteMode      WriteMode `json:"write_mode,omitempty"`
-	BatchSize      int       `json:"batch_size,omitempty"`
-	BatchSizeBytes int       `json:"batch_size_bytes,omitempty"`
-	Spec           any       `json:"spec,omitempty"`
+	Name           string      `json:"name,omitempty"`
+	Version        string      `json:"version,omitempty"`
+	Path           string      `json:"path,omitempty"`
+	Registry       Registry    `json:"registry,omitempty"`
+	WriteMode      WriteMode   `json:"write_mode,omitempty"`
+	MigrateMode    MigrateMode `json:"migrate_mode,omitempty"`
```
Comment (erezrokah) [inline specs/destination.go:22]: This can also be a boolean, just found it easier to communicate the purpose using strings

**Review by hermanschaaf — APPROVED** [review]
This LGTM; are we also planning on supporting a `--migrate forced` (or similar) flag for the CLI? Might make it a bit easier for users doing things interactively

**Comment by hermanschaaf** [comment]
We're also not using the setting anywhere yet - is that because it needs to be implemented in each destination? I think we'll need to pass the mode in to the `Migrate` call so that destinations can act on it?

**Review by disq — APPROVED** [review]

**Comment by erezrokah** [comment]
> a `--migrate forced` (or similar) flag for the CLI? Might make it a bit easier for users doing things interactively

So a CLI will we have to "take" for all specs, where a spec config only "takes" on the specific spec. We can consider adding it but I don't think we can similar flags (that override spec configuration(s))

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. One nit:

Can we move migrate enum code to `migrate.go` as looks like our enums are growing and maybe also add a small sanity test for marshal/unmarsahl.

**Comment by erezrokah** [comment]
> Can we move migrate enum code to `migrate.go` as looks like our enums are growing and maybe also add a small sanity test for marshal/unmarsahl.

Sure, good point. Let me know if https://github.com/cloudquery/plugin-sdk/pull/604/commits/1ff6062f3821a20e6a7e67f1aee6e026288b2cbb is what you had in mind

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**